cedar-policy-core/src/extensions/ipaddr.rs

```diff
@@ -242,15 +242,45 @@ fn contains_at_least_two(s: &str, c: char) -> bool {
     let idx = s.find(c);
     match idx {
         Some(i) => {
-            // PANIC SAFETY `i` is guaranteed to be < `s.len()`, so this won't panic
+            // For this slicing operation not to panic, two preconditions must be met:
+            // 1) `i + 1` must be <= `s.len()`
```

Comment:
```suggestion
            // 1) `i + c.len_utf8()` must be <= `s.len()`
```

which is less obviously correct